test/stablex/batch_exchange.js

```diff
@@ -330,15 +306,49 @@ contract("BatchExchange", async accounts => {
         solution.owners,
         solution.touchedOrderIds,
         solution.volumes,
-        solution.prices,
-        solution.tokenIdsForPrice,
+        solution.prices.slice(1),
+        solution.tokenIdsForPrice.slice(1),
```

Comment:
I think this slicing should move to `solutionSubmissionParams` instead of manually slicing every time `submitSolution` is called. After all, the point of `solutionSubmissionParams` is to generate the `submitSolution` arguments for a solution.

Reply:
I started by doing that, but as explained in the description we will find that many calculations and assertions depend on having this. Of course I wouldn’t be opposed to the necessary changes at all, but figured this was a good starting point.

Reply:
I don't think that solution generation in `generate.js` should change, just the function that extracts `submitSolution` parameters from an already generated solution (specifically [here](https://github.com/gnosis/dex-contracts/blob/c018721125a135d614aec0a5879a69279729520b/test/resources/examples/generate.js#L217)). If you mean the test assertions, I think updating them for every test is cleaner than slicing for every test.

Reply:
Agreed. Sorry for the confusion.